### PR TITLE
Increase net.tcp.py coverage

### PR DESCRIPTION
**Pull request by ujjwal96**
This PR increases net.tcp.py coverage to 100% making coverage of mitmproxy/net/ to 100%

### CODE REVIEW

File: mitmproxy/net/tcp.py [hunk 1]
```diff
@@ -849,7 +849,7 @@ def __init__(self, address):
         self.__is_shut_down = threading.Event()
         self.__shutdown_request = False
 
-        if self.address == 'localhost':
+        if self.address[0] == 'localhost':
```
Comment (mhils) [inline mitmproxy/net/tcp.py]: Good catch 👍

File: test/full_coverage_plugin.py [hunk 2]
```diff
@@ -59,6 +60,12 @@ def pytest_runtestloop(session):
     if os.name == 'nt':
         cov.exclude('pragma: windows no cover')
 
+    if sys.platform == 'darwin':
+        cov.exclude('pragma: osx no cover')
+
+    if 'OPENSSL_OLD' in os.environ:
+        cov.exclude('pragma: alpn no cover')
```
Comment (Kriechi) [inline test/full_coverage_plugin.py]: These patterns use a name to indicate *when* the code is not being covered.
Following this idea, `alpn no cover` would mean, code is not covered when ALPN is present.
I think a better name would be `pragma: missing-alpn no cover` or `pragma: openssl-old no cover`.

File: mitmproxy/net/tcp.py (hunk 1 above)
Comment (ujjwal96) [inline mitmproxy/net/tcp.py]: Now I see the importance of test cases and code coverage :smile:

**Comment by Kriechi** [comment]
Almost ready - please solve the conflict first.

**Pull request merged**